crates/apollo-compiler/src/validation/diagnostics.rs

```diff
@@ -0,0 +1,820 @@
+use crate::ast;
+use crate::ast::DirectiveLocation;
+use crate::ast::Name;
+use crate::ast::Type;
+use crate::ast::Value;
+use crate::diagnostic::CliReport;
+use crate::diagnostic::NodeLocation;
+use crate::Node;
+use std::fmt;
+use thiserror::Error;
+
+#[derive(Debug, Error, Clone, PartialEq, Eq)]
+pub(crate) struct ValidationError {
+    pub location: Option<NodeLocation>,
+    pub data: Box<DiagnosticData>,
```

Comment:
It's because ValidationError is sometimes used in `Result<(), ValidationError>` and then it affects the size of the result type. Though i'm not sure if it's actually helpful in practice to box this or if it's actually worse just to satisfy clippy.

Reply:
Eh, it's smaller now than it was, and even clippy is happy with it, so no reason to do it anymore (if there even was one).